framework/packages/abstract-interface/src/migrate.rs

```diff
@@ -114,26 +133,130 @@ impl<T: CwEnv> Abstract<T> {
         if ::proxy::contract::CONTRACT_VERSION != versions[1].module.info.version.to_string()
             && self.account.proxy.upload_if_needed()?.is_some()
         {
-            modules_to_register.push((
+            accounts_to_register.push((
                 self.account.proxy.as_instance(),
                 ::proxy::contract::CONTRACT_VERSION.to_string(),
             ));
         }
 
-        if !modules_to_register.is_empty() {
+        if !accounts_to_register.is_empty() {
             self.version_control
-                .register_account_mods(modules_to_register)?;
+                .register_account_mods(accounts_to_register)?;
             has_migrated = true
         }
 
-        if self.ibc.migrate_if_version_changed()? {
-            has_migrated = true
+        if self.ibc.deploy_or_migrate_if_version_changed(self)? {
+            has_migrated = true;
+
+            natives_to_register.push((
+                self.ibc.client.as_instance(),
+                ::ibc_client::contract::CONTRACT_VERSION.to_string(),
+            ));
+            natives_to_register.push((
+                self.ibc.host.as_instance(),
+                ::ibc_host::contract::CONTRACT_VERSION.to_string(),
+            ));
         }
 
+        self.version_control.register_natives(natives_to_register)?;
         self.version_control.approve_any_abstract_modules()?;
 
         Ok(has_migrated)
     }
+
+    /// Update native modules in version control  
+    pub fn update_natives_in_version_control(&self) -> Result<(), crate::AbstractInterfaceError> {
```

Comment:
The logic here is very similar to `fn migrate_if_version_changed`, is there any way to merge let them share a function?

Reply:
This function checks module in VC, while `migrate_if_version_changed` checks cw2 of contracts.

So this function is supposed to be used just to hotfix outdated versions of native contracts in Version Control when cw2 != module version in VC